### DISCUSSION TITLE
Does this still work ?

### DISCUSSION THREAD

**Discussion by muzzak123** [NickM-27/swatch#56] [General]
Try as I may I cannot get this to work under 
Home Assistant 2023.7.1
Supervisor 2023.07.1
Operating System 10.3
Frontend 20230705.1 - latest
Frigate works great but Swatch doesnt seem to work at all.
Is it still supported ? and does it work under the newer versions of HA after the DB schema change https://developers.home-assistant.io/blog/2023/01/02/db-schema-v32/

**Comment by NickM-27** [comment]
Works fine for me, what is not working exactly?

**Comment by muzzak123** [comment]
Well when I start it the log shows a number of exceptions and looks like this:

[2023-07-18 21:46:07,683]: INFO - Starting SwatchApp
[2023-07-18 21:46:07,689]: INFO - Importing SwatchApp Config
[2023-07-18 21:46:07,689]: INFO - Verified SwatchApp Config
Starting migrations
[2023-07-18 21:46:07,728]: INFO - Starting migrations
There is nothing to migrate
[2023-07-18 21:46:07,751]: INFO - There is nothing to migrate
[2023-07-18 21:46:07,751]: INFO - Starting Auto Detection for front_hd
[2023-07-18 21:46:07,752]: INFO - Starting snapshot cleanup
[2023-07-18 21:46:07,752]: INFO - Starting Detection Cleanup
Exception in thread Thread-2:
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
  File "/usr/local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 449, in _make_request
    six.raise_from(e, None)
  File "<string>", line 3, in raise_from
  File "/usr/local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 444, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/local/lib/python3.10/http/client.py", line 1374, in getresponse
    response.begin()
  File "/usr/local/lib/python3.10/http/client.py", line 318, in begin
    version, status, reason = self._read_status()
  File "/usr/local/lib/python3.10/http/client.py", line 287, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
http.client.RemoteDisconnected: Remote end closed connection without response
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/requests/adapters.py", line 440, in send
    resp = conn.urlopen(
  File "/usr/local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 787, in urlopen
    retries = retries.increment(
  File "/usr/local/lib/python3.10/site-packages/urllib3/util/retry.py", line 550, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/usr/local/lib/python3.10/site-packages/urllib3/packages/six.py", line 769, in reraise
    raise value.with_traceback(tb)
  File "/usr/local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 703, in urlopen
    httplib_response = self._make_request(
  File "/usr/local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 449, in _make_request
    six.raise_from(e, None)
  File "<string>", line 3, in raise_from
  File "/usr/local/lib/python3.10/site-packages/urllib3/connectionpool.py", line 444, in _make_request
    httplib_response = conn.getresponse()
  File "/usr/local/lib/python3.10/http/client.py", line 1374, in getresponse
    response.begin()
  File "/usr/local/lib/python3.10/http/client.py", line 318, in begin
    version, status, reason = self._read_status()
  File "/usr/local/lib/python3.10/http/client.py", line 287, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "/opt/swatch/swatch/detection.py", line 122, in run
    result: Dict[str, Any] = self.image_processor.detect(
  File "/opt/swatch/swatch/image.py", line 122, in detect
    img_bytes = requests.get(image_url).content
  File "/usr/local/lib/python3.10/site-packages/requests/api.py", line 75, in get
    return request('get', url, params=params, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/requests/sessions.py", line 529, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.10/site-packages/requests/sessions.py", line 645, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/requests/adapters.py", line 501, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))


The image URL I was using for the snapshot config is http://192.168.0.102:8123/local/bins.jpg with a file in my HA /config/www directory called bins.jpg
I suspect this is not correct?

**Reply by NickM-27** [reply]
Right, I don't think it is accessible that way via a get call in HA. Only www is accessible in that way afaik

**Comment by muzzak123** [comment]
Ok so what should the URL be for a file on my local Home Assistant ?
Your example config shows 
url: "http://ip.ad.dr.ess/jpg"
Could you elaborate a bit more ? Perhaps with some examples. 
And maybe some examples for the API that include an example full path to an image. 
Your example of "http://some_camera_image.jpg" is a bit hard to follow and I'm even sure if that's a valid URL ?
 
Also my home assistant uses an external URL https://myname.dns.com:8213/
When replace the snapshot config URL with an external URL I don't get the previous errors.
But when I click on OPEN WEB GUI it seems to go to
http://myname.dns.com:4500/ which of course results in "ERR_CONNECTION_REFUSED"

**Reply by NickM-27** [reply]
It needs to be a url that when you open it in a web browser in incognito mode it can show the image without logging in

**Reply by muzzak123** [reply]
I did use an URL that met those requirements and it still doesnt work.
And what about the fact that the OPEN WEB GUI won't open the correct HTTPS URL for HA ?
You havent really addressed any of the issues I mentioned so I'm sensing a distinct lack of interest. It's a shame because it could have been quite a good product. I'm flagging is as too underdeveloped to be of any use.

**Reply by NickM-27** [reply]
> And what about the fact that the OPEN WEB GUI won't open the correct HTTPS URL for HA ?

ingress is not supported so that won't work

> You havent really addressed any of the issues I mentioned so I'm sensing a distinct lack of interest. It's a shame because it could have been quite a good product. I'm flagging is as too underdeveloped to be of any use.

swatch is a very simple project that I use to request snapshots from frigate to see if my trash cans are currently out on the curb. For this purpose it works well. Other users have gotten similar use out of it. I don't have plans to invest large amounts of time into it because I don't see it being very popular and I much more enjoy contributing to frigate.

**Comment by muzzak123** [comment]
That's kinda different to what you say here:
https://www.reddit.com/r/homeassistant/comments/ujqp6c/comment/i7r2b69/

> It is still early days and I have a lot of plans for ways to refine / improve, but if anyone wants to check it out that would be awesome! Happy to accept bugs / feature requests on GitHub and any issues please let me know!<

May I suggest you update your github to reflect your above feelings and perhaps mention what is not supported, so that others don't fall into the same trap I did and wast a lot of time trying to get it working, whilst under the impression it WAS going to be developed further (**a lot of plans for ways to refine / improve)**
Also why don't you put your talents to good work and add it as an option within Frigate. It has many more uses than just identifying if your bins are out.
